R/mod_packageDependencies.R

```diff
@@ -0,0 +1,336 @@
+#' @importFrom rlang is_empty
+#' @importFrom glue glue
+#' @importFrom DT renderDataTable formatStyle
+#' @importFrom stringr str_replace word
+#' @importFrom shiny tagList showModal removeModal
+#' @importFrom shinyjs click
+#' @importFrom deepdep deepdep plot_dependencies
+#' @importFrom formattable as.datatable formattable formatter style
+#' @importFrom cli cli_progress_message cli_progress_bar cli_progress_update cli_progress_done
+#' 
+#' @keywords internal
+#' 
+packageDependenciesServer <- function(id, selected_pkg, user, changes, parent) {
+  moduleServer(id, function(input, output, session) {
+    ns <- session$ns
+    
+    metric_wts_df <- dbSelect("SELECT id, name, weight FROM metric")
+    metric_weights <- metric_wts_df$weight
+    names(metric_weights) <- metric_wts_df$name
+    
+    loaded2_db <- reactive({
+      dbSelect('SELECT name FROM package')$name
+    })
+    
+    get_versnScore <- function(pkg_name, id) {
+      
+      riskmetric_assess <-
+        riskmetric::pkg_ref(pkg_name,
+                            source = "pkg_cran_remote",
+                            repos = c("https://cran.rstudio.com")) %>%
+        dplyr::as_tibble() %>%
+        riskmetric::pkg_assess()
+      
+      riskmetric_score <-
+        riskmetric_assess %>%
+        riskmetric::pkg_score(weights = metric_weights)
+      
+      cli::cli_progress_update(id = id)
+      
+      return(list(name = riskmetric_assess$package, version = riskmetric_assess$version, score = round(riskmetric_score$pkg_score,2) ))
+    }
+    
+    # used for adding action buttons to data_table
+    shinyInput <- function(FUN, len, id, ...) {
+      inputs <- character(len)
+      for (i in seq_len(len)) {
+        inputs[i] <- as.character(FUN(paste0(id, i), ...))
+        # change icon from arrow to upload if not loaded into db yet
+        pkg_name <- pkg_df()[i, 3] %>% pull()
+        if (!pkg_name %in% loaded2_db()) {
+          inputs[i] <- gsub("fas fa-arrow-right fa-regular", "fas fa-upload fa-solid", inputs[i])
+        }
+      }
+      inputs
+    }
+    
+    depends <- reactive({
+      req(selected_pkg$name() != "-")
+      riskmetric::pkg_ref(selected_pkg$name()) %>% 
+        riskmetric::assess_dependencies() 
+    })
+    
+    revdeps <- reactive({
+      req(selected_pkg$name() != "-")
+      riskmetric::pkg_ref(selected_pkg$name()) %>% 
+        riskmetric::assess_reverse_dependencies() 
+    })
+    
+    dd <- eventReactive(pkg_df(), {
+      x <- deepdep::deepdep(selected_pkg$name(), depth = 2, dependency_type = c("Depends", "Imports", "LinkingTo"))
+      if (nrow(x) == 0) {
+        x <- data.frame(origin = selected_pkg$name(),
+                        name = pull(pkg_df()[1, 3]), version = NA_character_, type = "Imports",
+                        origin_level = 0L, dest_level = 1L)
+        attributes(x)$class <- c("deepdep", "data.frame")
+        attributes(x)$package_name <- selected_pkg$name()
+      }
+      return(x)
+    })
```

Comment:
I see now that you were creating the `deepdep` object in the code, not calling by package name in the plotting function.

Reply:
Yes, I ended up creating an empty df in case `deepdep::deepdep()` doesn't return any rows.